### ISSUE TITLE
Incorrectly render lines overflow the chart.

### ISSUE THREAD

**Issue by liyuanqiu** [apache/incubator-echarts#9200]
### One-line summary [问题简述]
当使用dataZoom进行缩放时，线条会溢出图表。详见下面给出的JSFiddle

### Version & Environment [版本及环境]
+ ECharts version [ECharts 版本]: 4.10
+ Browser version [浏览器类型和版本]: 69
+ OS Version [操作系统类型和版本]: macOS

### Expected behaviour [期望结果]
进行缩放时，不要将线条渲染到图表之外。

### ECharts option [ECharts配置项]

```javascript
option = {
       xAxis: {
          type: 'time',
          min: 1462294400000,
          max: 1462294550000,
        },
        yAxis: {
          type: 'category',
          data: ['A', 'B'],
        },
        dataZoom: [{
          type: 'slider',
          xAxisIndex: [0],
        }],
        series: [{
          data: [{
            coords: [
              [1462294400000, 'A'],
              [1462294500000, 'A'],
            ],
          }, {
            coords: [
              [1462294450000, 'B'],
              [1462294550000, 'B'],
            ],
          }],
          type: 'lines',
          coordinateSystem: 'cartesian2d',
          silent: false,
        }]
}

```

### Other comments [其他信息]

http://jsfiddle.net/62t3d9bf/

**Comment by Ovilia** [comment]
We should probably put dataZoom overflow bug at a higher priority. @100pah 
See also #8935.

**Comment by Ovilia** [comment]
Check if #10748 is the same problem.

**Comment by liyuanqiu** [comment]
> Check if #10748 is the same problem.

Maybe the same problem.

**Comment by Ovilia** [comment]
#11010 may also be related.

**Comment by pissang** [comment]
Added `clip` option for lines series in the commit https://github.com/apache/incubator-echarts/commit/94dfc9284f12cba56b7524dcf60776d0c39dbe79 . It will be included in version 4.4.0

Here is the screenshot after fixed.
![image](https://user-images.githubusercontent.com/841551/65024553-adaa7880-d967-11e9-8f86-5954a0199168.png)
